docker_publisher.py

```diff
@@ -0,0 +1,623 @@
+        self.url = url
+
+    def currentVersion(self):
+        return self.version
+
+    def getDockerImage(self, callback):
+        """Download the tar and extract it"""
+        with tempfile.NamedTemporaryFile() as tar_file:
+            tar_file.write(requests.get(self.url).content)
+            with tempfile.TemporaryDirectory() as tar_dir:
+                # Extract the .tar.xz into the dir
+                subprocess.call("tar -xaf '%s' -C '%s'" % (tar_file.name, tar_dir), shell=True)
+                return callback(tar_dir)
+
+
+class DockerImageFetcherOBS(DockerImageFetcher):
+    """Uses the OBS API to access the build artifacts.
+    Url has to be https://build.opensuse.org/public/build/<project>/<repo>/<arch>/<pkgname>"""
+    def __init__(self, url):
+        self.url = url
+
+    def _getFilename(self):
+        """Return the name of the binary at the URL with the filename ending in
+        .docker.tar.xz."""
+        binarylist_req = requests.get(self.url)
+        binarylist = xml.fromstring(binarylist_req.content)
+        for binary in binarylist.xpath("binary/@filename"):
+            if binary.endswith(".docker.tar.xz"):
+                return binary
+
+        raise DockerFetchException("No docker image built in the repository")
+
+    def currentVersion(self):
+        """Return {version}-Build{build} of the docker file."""
+        filename = self._getFilename()
+        return re.match(".*((-[^-]+){2})\.docker\.tar\.xz", filename).group(1)[1:]
+
+    def getDockerImage(self, callback):
+        """Download the tar and extract it"""
+        filename = self._getFilename()
+        with tempfile.NamedTemporaryFile() as tar_file:
+            tar_file.write(requests.get(self.url + "/" + filename).content)
+            with tempfile.TemporaryDirectory() as tar_dir:
+                # Extract the .tar.xz into the dir
+                subprocess.call("tar -xaf '%s' -C '%s'" % (tar_file.name, tar_dir), shell=True)
+                return callback(tar_dir)
+
+
+class DockerImageFetcherRepo(DockerImageFetcher):
+    """This can be used when the image is wrapped into an RPM and released as
+    part of the main repository, as it is the case for Tumbleweed.
+    The version equals the version of the product in the repository, determined
+    by the versioned_redir URL redirection target."""
+    def __init__(self, versioned_redir, repourl, packagename, arch):
+        self.versioned_redir = versioned_redir
+        self.repourl = repourl
+        self.packagename = packagename
+        self.arch = arch
+
+    def currentVersion(self):
+        # For TW we ask the mirrorbrain server about the -Current redirection target
+        meta4_xml = requests.get(self.versioned_redir + ".meta4")
+        meta4 = xml.fromstring(meta4_xml.content)
+        filename = meta4.xpath("//m:metalink//m:file//@name", namespaces={'m': 'urn:ietf:params:xml:ns:metalink'})[0]
+        return re.search('Snapshot(\d+)-', filename).group(1)
```

Comment:
nit: perhaps use regex string `r"pattern"` to avoid issues in future

Reply:
Indeed, fixed.